flask_api/app.py

```diff
@@ -89,4 +136,89 @@ def delete_list(list_id):
         return jsonify('Unauthorized'), 401
 
     db_table().delete_item(Key={'userId': user_id, 'listId': list_id})
-    return jsonify()
\ No newline at end of file
+    return jsonify()
+
+def fibonacci(index):
+    """
+    Recursive function that calculates Fibonacci sequence.
+    :param index: the n-th element of Fibonacci sequence to calculate.
+    :return: n-th element of Fibonacci sequence.
+    """
+
+    if index <= 1:
+        return index
+    return fibonacci(index - 1) + fibonacci(index - 2)
+
+
+@app.route('/fib/<int:n>')
+def fib(n):
+    
+    for _ in range(10):
+         result = { 'The ' + str(n) + ' fibonacci number is' : fibonacci(n)}
+    return jsonify(result)
+
+
+@app.route('/wavefile')
+def wave_api():
+
+    if random.uniform(0, 100) > 65:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *wavefile*. The resource is allocated by call *wave.open*. Execution paths that do not contain closure statements were detected. Close *wavefile* in a try-finally block to prevent the resource leak.